File: Accounting.DATA/Entity/Payment.cs
using Accounting.DATA.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace Accounting.DATA.Entity
{
    public class Payment
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public Guid CustomerId { get; set; }

        public Guid? InvoiceId { get; set; }
        public Guid? InstallmentId { get; set; }
        public Guid? CheckId { get; set; }
        public Guid? ExpenseId { get; set; }
        
        public double Amount { get; set; }
        public Guid? BankId { get; set; }
        public Guid? SafesId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string DocNo { get; set; }
        public DateTime? Date { get; set; }
        public PaymentType Type { get; set; }

        public bool Status { get; set; }
        public DateTime? CreatedOn { get; set; }
        public DateTime? UpdatedOn { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime? DeletedDate { get; set; }
    }
}

File: Accounting/wwwroot/js/Payment.js
function GetPaymentMethod() {
    var paymentMethod = $('#paymentMethod').val();

    if (paymentMethod == 1 || paymentMethod == 2) {
        $('#banks').attr("hidden", false);
        $('#creditCard').attr("hidden", true);
    }
    if (paymentMethod == 3 || paymentMethod == 4) {
        $('#creditCard').attr("hidden", false);
        $('#banks').attr("hidden", true);
    }
}

function Create() {
    var payment = {
        CustomerId: $('#CustomerId').val(),
        BankId: $('#bank').val(),
        SafesId: $('#safe').val(),
        Amount: $('#amount').val(),
        Description: $('#description').val(),
        Date: $('#ProcessDate').val(),
        Type: $('#Type').val(),
    }

    var InvoiceId = $('#InvoiceId').val();
    var InstallmentId = $('#InstallmentId').val();
    var CheckId = $('#CheckId').val();

    if (InvoiceId != null) {
        payment.InvoiceId = InvoiceId;
    }
    if (InstallmentId != null) {
        payment.InstallmentId = InstallmentId;
    }
    if (CheckId != null) {
        payment.CheckId = CheckId;
    }

    $.ajax({
        url: "/Payment/Create",
        data: {
            "payment": payment
        },
        type: "POST",

        success: function (response) {
            if (response.status == true) {

                toastr.success(response.message);

                setTimeout(function () {

                }, 700);
            }
            else {
                toastr.error(response.message);
            }
        }
    })


}